packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,237 @@
+   * @param notificationRuleArn Notification rule ARN (i.e. arn:aws:codestar-notifications:::notificationrule/01234abcde)
+   */
+  public static fromNotificationRuleArn(scope: Construct, id: string, notificationRuleArn: string): INotificationRule {
+    class Import extends Resource implements INotificationRule {
+      readonly notificationRuleArn = notificationRuleArn;
+
+      public addTarget(_target?: INotificationRuleTarget) {
+        return false;
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * @attribute
+   */
+  readonly notificationRuleArn: string;
+
+  /**
+   * The source config of notification rule
+   */
+  readonly source: NotificationRuleSourceConfig;
+
+  /**
+   * The target config of notification rule
+   */
+  readonly targets: NotificationRuleTargetConfig[] = [];
+
+  /**
+   * The events of notification rule
+   */
+  readonly events: string[] = [];
+
+  constructor(scope: Construct, id: string, props: NotificationRuleProps) {
+    super(scope, id);
+
+    this.source = props.source.bind(this);
+
+    if (props.events) {
+      this.addEvents(props.events);
+    }
+
+    if (props.target) {
+      this.addTarget(props.target);
+    }
+
+    const resource = new CfnNotificationRule(this, 'Resource', {
+      name: props.notificationRuleName || this.generateName(),
+      status: (props?.enabled === false) ? Status.DISABLED : Status.ENABLED,
+      detailType: props.detailType || DetailType.FULL,
+      targets: this.targets,
+      eventTypeIds: this.events,
+      resource: this.source.sourceArn,
+    });
+
+    this.notificationRuleArn = resource.ref;
+  }
+
+  /**
+   * Adds target to notification rule
+   * @param target The SNS topic or AWS Chatbot Slack target
+   */
+  public addTarget(target?: INotificationRuleTarget): boolean {
+    if (!target) {
+      return false;
+    }
+
+    this.targets.push(target.bind(this));
+    return true;
+  }
+
+  /**
+   * Adds events to notification rule
+   *
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#events-ref-pipeline
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#events-ref-buildproject
+   * @param events The list of event types for AWS Codebuild and AWS CodePipeline
+   */
+  private addEvents(events: string[]): void {
+    events.forEach((event) => {
+      if (this.events.includes(event)) {
+        return;
+      }
+
+      this.events.push(event);
+    });
+  }
+
+  /**
+   * The name length generated by id must short than 64 characters
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codestarnotifications-notificationrule.html#cfn-codestarnotifications-notificationrule-name
+   * @returns string
+   */
+  private generateName(): string {
+    const name = Names.uniqueId(this);
+    if (name.length > 64) {
+      return name.substring(0, 32) + name.substring(name.length - 32);
+    }
+
+    return name;
```

Comment:
Sorry, what's happening here?

Like I said - take the last 64 characters of the unique ID. [Like this](https://github.com/aws/aws-cdk/blob/252dfa2f84f24ef57ab632e8ee5092544c850a5f/packages/%40aws-cdk/aws-rds/lib/database-secret.ts#L93-L93).

Reply:
Already adjusted.